pulp_ansible/app/galaxy/views.py

```diff
@@ -154,18 +155,25 @@ def post(self, request, path):
         )
         serializer.is_valid(raise_exception=True)
 
-        artifact = Artifact.init_and_validate(serializer.validated_data["file"])
-        artifact.save()
+        temp_file = serializer.validated_data["file"]
```

Comment:
What path does this get? Is it alread inside `/var/lib/pulp`/?

Reply:
`/var/lib/pulp/tmp`

Reply:
What happens if we use `temp_file_path = serializer.validated_data["file"]` rather than using `shutil` at all? I suspect it may not work, but I wanted to ask.

Reply:
it was my original approach,
```python
serializer.validated_data["file"] == PulpTemporaryUploadedFile == tempfile.NamedTemporaryFile
```
By the time the task is executed, the file is gone

Reply:
I see, yes that is what I was concerned about. Then just using `temp_file_path` as I was thinking won't work.

There is a small (although unlikely) issue with your code building the name instead of having tempfile pick the file name instead. I believe `tempfile` contains code which guarantees to be free of race conditions. Here's the gist of what I was thinking.

```
with tempfile.NamedTemporaryFile(dir=settings.WORKING_DIRECTORY, delete=False) as temp_import_file:
    shutil.copyfileobj(serializer.validated_data["file"], temp_import_file)
    temp_file_path = temp_import_file.name   # I forget the attribute name but it's like name or named
```

What do you think about this?

Also the code could be deduplicated in another method named `_create_temp_file` on your exiting Mixin, which might be nice too.

Reply:
it looks good, I will work on it